Automattic/jetpack
# Add Anchor.fm Badge Inserter

Fixes 317-gh-Automattic/dotcom-manage

#### Changes proposed in this Pull Request:
* adds a new "block" into extensions so we can control the availability of this functionality (once launched, it will be available for all plans)
* look for GET params when opening the post editor and if they are found and matched with episode details:
  - a link badge will be inserted at the top of the post
  - post meta will be stored (`jetpack_anchor_podcast`, `jetpack_anchor_episode`, and `jetpack_anchor_spotify_show`)

#### Jetpack product discussion
pbAPfg-PF-p2

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
- Activate the beta blocks:
  - If testing on a JN site, go to Settings > Jetpack Constants and turn on the `JETPACK_BETA_BLOCKS` option.
  - If testing locally, add a new plugin to `docker/mu-plugins` with `define( 'JETPACK_BETA_BLOCKS', true );`.
- Make a testing post in the block editor and save it.
- Open the post for editing and add three GET params to the URL — `anchor_podcast`, `anchor_episode`, and `spotify_show_url`:
  - for convenience, you can use: `&anchor_podcast=22b6608&anchor_episode=e324a06c-3148-43a4-85d8-34c0d8222138&spotify_show_url=https://open.spotify.com/show/6HTZdaDHjqXKDE4acYffoD?si=EVfDYETjQCu7pasVG5D73Q`
- There should be a badge added on top of the post and you should be able to confirm post meta being set by making a `GET` request to the `/wp/v2/sites/:site/posts/:postId` endpoint:

<img width="657" alt="Screenshot 2020-12-01 at 16 54 38" src="https://user-images.githubusercontent.com/156676/100763863-04c62d00-33f6-11eb-8e9f-a7b7bf6ff0ab.png">

<img width="1263" alt="Screen Shot 2020-12-08 at 16 14 18" src="https://user-images.githubusercontent.com/1233880/101503117-0f427280-3972-11eb-9389-018d2923e735.png">


- Save the post and reload the editor with the same GET params.
- Make sure the badge is not duplicated.


#### Proposed changelog entry for your changes:
—
